packages/react-router/src/router.ts

```diff
@@ -2528,6 +2528,8 @@ export class Router<
                             loaderData,
                           })
                           const meta = headFnContent?.meta
+                          const links = headFnContent?.links
```

Comment:
I don't understand why this worked in `createFileRoute` but in in `createRoute`. can you explain this please?

Reply:
Sorry, I haven't had time to fully evaluate this but agree it didn't seem immediately apparent why this worked using createRootRoute and createFileRoute vs createRoute. I have to assume they go through the same path and it seems file route calls createRoute under the hood... I will explore this more but as I say I haven't yet found the time to give this an in depth diagnosis beyond the initial findings.